### PR TITLE
s160 - working habitat vectorization

### PR DESCRIPTION
**Pull request by thanrobinson**

### CODE REVIEW

**Comment by thanrobinson** [comment]
[eff pot hab mvp](https://trello.com/c/P7aSRn3u/160-eff-pot-hab-mvp)

File: src/task_scl_eff_pot_hab.py [hunk 1]
```diff
@@ -3,180 +3,200 @@
...
 
 class SCLPolygons(SCLTask):
     ee_rootdir = "projects/SCL/v1"
+    scale = 300
     inputs = {
         "structural_habitat": {
             "ee_type": SCLTask.IMAGECOLLECTION,
             "ee_path": "projects/SCL/v1/Panthera_tigris/geographies/Sumatra/hab/structural_habitat",
-            "maxage": 1,  # years
+            "maxage": 5,  # years
```
Comment (saanobhaai) [inline src/task_scl_eff_pot_hab.py]: This should stay `1` shouldn't it? We do have annual images for the years we need, for Sumatra...

File: src/task_scl_eff_pot_hab.py [hunk 2]
```diff
@@ -3,180 +3,200 @@
...
         "hii": {
             "ee_type": SCLTask.IMAGECOLLECTION,
             "ee_path": "projects/HII/v1/sumatra_poc/hii",
-            "maxage": 1,
+            "maxage": 30,
+        },
+        "probability": {
+            "ee_type": SCLTask.IMAGECOLLECTION,
+            "ee_path": "projects/SCL/v1/Panthera_tigris/geographies/Sumatra/hab/probability",
+            "maxage": 30,
```
Comment (saanobhaai) [inline src/task_scl_eff_pot_hab.py]: I think this should be `"maxage": 1` too

File: src/task_scl_eff_pot_hab.py [hunk 3]
```diff
@@ -3,180 +3,200 @@
...
         },
         "historic_range": {
             "ee_type": SCLTask.IMAGE,
             "ee_path": "projects/SCL/v1/Panthera_tigris/source/Inputs_2006/hist",
-            "maxage": 1 / 365,
+            "maxage": 20,
```
Comment (saanobhaai) [inline src/task_scl_eff_pot_hab.py]: If you include `"static": 20` probably best to remove the maxage property -- see below inputs as well

File: src/task_scl_eff_pot_hab.py [hunk 4]
```diff
@@ -3,180 +3,200 @@
...
+            self.hii.lt(self.thresholds["hii"])
+        ).selfMask()
+        excluded_habitat = connected_structural_habitat.updateMask(
+            self.hii.gte(self.thresholds["hii"])
+        ).selfMask()
+        effective_potential_habitat = potential_habitat.addBands(
+            excluded_habitat
+        ).rename(["eff_pot_hab", "excl_hab"])
+
+        # probability_calc = self.elevation.expression(
```
Comment (saanobhaai) [inline src/task_scl_eff_pot_hab.py]: Remove commented lines

File: src/task_scl_eff_pot_hab.py [hunk 5]
```diff
@@ -3,180 +3,200 @@
...
+        #     "1 - (2.71828 ** (-(2.71828 ** (- 0.069 - 0.044 * ((dem - 639) / 592) - 0.12 * ((hii - 7.72) / 5.47)))))",
+        #     {"dem": self.elevation, "hii": self.hii},
+        # )
+
+        high_probability = self.probability.gte(self.thresholds["probability"]).multiply(
```
Comment (saanobhaai) [inline src/task_scl_eff_pot_hab.py:116]: I still don't really understand the `multiply` calls -- why multiply by 2, or even more why multiply by 1? Maybe you can indicate in a comment?

File: src/task_scl_eff_pot_hab.py [hunk 6]
```diff
@@ -3,180 +3,200 @@
...
+            .add(effective_potential_habitat.select(1).unmask(0).multiply(2))
+            .selfMask()
+            .remap([1, 2], [2, 3])
+        )
 
-    def calc(self):
-        self.delineate_polygons()
+        hii_ecoregion = self.ecoregion_country.add(hii_sum)
+
+        minPatchSize = self.density.remap(
```
Comment (saanobhaai) [inline src/task_scl_eff_pot_hab.py]: python convention is to use lowercase var names separated by underscores, not camelCase

File: src/task_scl_eff_pot_hab.py [hunk 7]
```diff
@@ -3,180 +3,200 @@
...
+            .combine(ee.Reducer.sum().unweighted(), "area_")
+            .combine(ee.Reducer.mode().unweighted(), "patch_")
+            .combine(ee.Reducer.mode().unweighted(), "stpStn_")
+        )
+        hii_grp_poly = hii_grp.reduceToVectors(
+            reducer=polygonReducer,
+            geometry=ee.Geometry.MultiPolygon(self.aoi).bounds(),
+            crs=self.crs,
+            scale=self.scale,
+            maxPixels=1e13,
```
Comment (saanobhaai) [inline src/task_scl_eff_pot_hab.py]: There's a `max_pixels` property in task_base as well, so you can use that like you are crs/scale

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Looks great @thanrobinson -- just a few minor comments

File: src/task_scl_eff_pot_hab.py (hunk 5 above)
Comment (thanrobinson) [reply src/task_scl_eff_pot_hab.py:116]: we want a single classified image of high and low probability. there are a number of means to get this.  we could use a image.where (which might be more clean). This was very quick/easy to code up.  image.gte creates a 0/1 binary image. same for image.lt.  If we just combine these two, we end up with an image where both high and low prob is classified as 1. the multiply by 1 is completely unnecessary. could also use a image.remap function.

File: src/task_scl_eff_pot_hab.py (hunk 1 above)
Comment (thanrobinson) [reply src/task_scl_eff_pot_hab.py]: the structural_habitat only goes through 2018, and the hii through 2017. So to run currently run task with no -d argument, the maxage needs to be >1 for the moment. I included comments in the code to remind us to change when full input collections are available

File: src/task_scl_eff_pot_hab.py (hunk 6 above)
Comment (thanrobinson) [reply src/task_scl_eff_pot_hab.py]: fixed these

File: src/task_scl_eff_pot_hab.py (hunk 7 above)
Comment (thanrobinson) [reply src/task_scl_eff_pot_hab.py]: changed to self.ee_max_pixels

File: src/task_scl_eff_pot_hab.py (hunk 4 above)
Comment (thanrobinson) [reply src/task_scl_eff_pot_hab.py]: removed

File: src/task_scl_eff_pot_hab.py (hunk 3 above)
Comment (thanrobinson) [reply src/task_scl_eff_pot_hab.py]: all static inputs have no maxage now

File: src/task_scl_eff_pot_hab.py (hunk 2 above)
Comment (thanrobinson) [reply src/task_scl_eff_pot_hab.py]: see comment to structural_hab maxage

**Pull request merged**